src/resolve/resolved-dns-packet.c

```diff
@@ -2570,6 +2571,81 @@ bool dns_packet_equal(const DnsPacket *a, const DnsPacket *b) {
         return dns_packet_compare_func(a, b) == 0;
 }
 
+int dns_packet_ede_rcode(DnsPacket *p, char **ret_ede_msg) {
+        assert(p);
+
+        _cleanup_free_ char *msg = NULL, *msg_escaped = NULL;
+        int ede_rcode = _DNS_EDNS_OPT_MAX_DEFINED;
+        int r;
+        const uint8_t *d;
+        size_t l;
+
+        if (!p->opt)
+                return -EINVAL;
```

Comment:
doesn't matter at all, but I think it would be nicer to return _DNS_EDE_RCODE_INVALID here, which is of course the exact same, but kinda gets the point across to the reader that this just means the rcode info is not valid here